sphinx/writers/latex.py

```diff
@@ -730,32 +667,9 @@ def babel_renewcommand(self, command, definition):
 
         return ('%s\\renewcommand{%s}{%s}%s\n' % (prefix, command, definition, suffix))
 
-    def babel_defmacro(self, name, definition):
-        # type: (unicode, unicode) -> unicode
-        if self.elements['babel']:
-            prefix = '\\addto\\extras%s{' % self.babel.get_language()
-            suffix = '}'
-        else:  # babel is disabled (mainly for Japanese environment)
-            prefix = ''
-            suffix = ''
-
-        return ('%s\\def%s{%s}%s\n' % (prefix, name, definition, suffix))
-
     def generate_numfig_format(self, builder):
         # type: (Builder) -> unicode
         ret = []  # type: List[unicode]
-        figure = self.builder.config.numfig_format['figure'].split('%s', 1)
-        if len(figure) == 1:
```

Comment:
I dropped this block on migration because `numfig_format` expects the value contains `'%s'`.

Reply:
Perhaps our [doc](https://www.sphinx-doc.org/en/master/usage/configuration.html?highlight=numfig_format#confval-numfig_format) should make it clearer that `%s` is mandatory. By the way for Section the `%s` is lacking in our [doc](https://www.sphinx-doc.org/en/master/usage/configuration.html?highlight=numfig_format#confval-numfig_format)